tutorials/Tutorial MIS & QAOA.ipynb

```diff
@@ -0,0 +1,394 @@
+    "^z + 1)/2$ counts the number of Rydberg excitations at position $i$. The interaction strength between two atoms decays as $|\\textbf{r}_i-\\textbf{r}_j|^{-6}$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# From a graph to an atomic register"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now illustrate how one can use Pulser and a Pasqal device to find the MIS of a UD-graph. Because the quantum platform is emulated in this notebook, we restrict the number of atoms to 5, just to show a proof-of-concept."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pos_to_graph(pos, d = Chadoq2.rydberg_blockade_radius(1)): #d = rydberg blockade\n",
+    "    edges=[]\n",
+    "    for n in range(len(pos)-1):\n",
+    "        for m in range(n+1, len(pos)):\n",
+    "            pwd = ((pos[m][0]-pos[n][0])**2+(pos[m][1]-pos[n][1])**2)**0.5\n",
+    "            #print('distance=%f'%pwd)\n",
+    "            if pwd < d:\n",
+    "                edges.append([n,m])                         # Below a given distance, vertices are connected\n",
+    "    #Now we use a igraph object to describe our graph. This module has several interesting built-in functions.\n",
+    "    g=igraph.Graph()\n",
+    "    g.add_vertices(len(pos))\n",
+    "    g.add_edges(edges)\n",
+    "    return g"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we create an atomic register with 5 atoms and the corresponding graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos = np.array([[0., 0.], [-4, -7], [4,-7], [8,6], [-8,6]])\n",
+    "G = pos_to_graph(pos) \n",
+    "qubits = dict(enumerate(pos))\n",
+    "reg = Register(qubits)\n",
+    "reg.draw()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import networkx as nx\n",
+    "A  = np.matrix([[0, 1, 1, 1, 1], [1, 0, 1, 0, 0], [1, 1, 0, 0, 0], [1, 0, 0, 0, 0], [1, 0, 0, 0, 0]])\n",
+    "G_nx = nx.from_numpy_matrix(A)\n",
+    "fig,ax1 = plt.subplots(figsize = (4,4))\n",
+    "nx.draw(G_nx,\n",
+    "        pos,\n",
+    "        edge_color='blue', \n",
+    "        ax=ax1,\n",
+    "        width=1,\n",
+    "        linewidths=1,\n",
+    "        node_size=250,\n",
+    "        node_color='g',\n",
+    "        alpha=1.,\n",
+    "        labels={node:node for node in G_nx.nodes()}\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This graph has two maximal independent sets: $(1,3,4)$ and $(2,3,4)$, respectively $'10100'$ and $'11000'$ in binary. "
```

Comment:
I actually put the opposite values (01011 and 00111) in the rest of the code. So here there has to be consistency.